sciencer/expanders/expand_by_references.py

```diff
@@ -0,0 +1,33 @@
+"""Expands paper by its references
+"""
+from typing import List
+from ..providers.provider import Provider
+from .expander import Expander
+from ..models import Paper
+
+
+class ExpandByReferences(Expander):
+    """Class that encapsulates the expansio of a paper by its references
+
+    Args:
+        Expander (_type_): _description_
+    """
+
+    def __init__(self) -> None:
+        super().__init__([])
+
+    def execute(self, papers: List[Paper], providers: List[Provider]) -> List[Paper]:
+        resulting_papers = set()
+
+        referenced_papers_ids = set()

```

Comment:
Just merge this `for` into the next. Unless you have an hidden agenda going on here, this just seems like a waste of space...

Reply:
The first for is used to identify all the referenced papers (without duplicates)
The second for is used to expand these referenced papers (avoiding redundant expansions)

Reply:
Ah. The duplicates part of the first for had eluded me.